standard/mock/mock_team_template.lua

```diff
@@ -0,0 +1,257 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Mock/TeamTemplate
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+-- Warning: This mock ignores the posibility of image/imagedark being unset/empty
+-- and never uses legacyimage/legacyimagedark for that reason
+
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local MockData = Lua.import('test_assets.team_template_data')
+
+local NOT_FOUND_ERROR = '<div class="error">No team template exists for name "${name}".</div>'
+local MISSING_INPUT_ERROR = 'bad argument #1 to "name" (string or number expected, got nil)'
+
+local mockTeamTemplate = {}
+
+-- temp copy the orig functions so we can retrieve them later again when tearing down the mock
+local _teamTemplate = mw.ext.TeamTemplate
+
+function mockTeamTemplate.setUp()
+	mw.ext.TeamTemplate = {
+		raw = mockTeamTemplate.raw,
+		raw_historical = mockTeamTemplate.raw_historical,
+		teamexists = mockTeamTemplate.teamexists,
+		team = mockTeamTemplate.team,
+		team2 = mockTeamTemplate.team2,
+		teamshort = mockTeamTemplate.teamshort,
+		team2short = mockTeamTemplate.team2short,
+		teambracket = mockTeamTemplate.teambracket,
+		teamicon = mockTeamTemplate.teamicon,
+		teamimage = mockTeamTemplate.teamimage,
+		teampage = mockTeamTemplate.teampage,
+		teampart = mockTeamTemplate.teampart,
+	}
+end
+
+function mockTeamTemplate.tearDown()
+	mw.ext.TeamTemplate = _teamTemplate
+end
+
+---@param teamtemplate string
+---@param date string|number?
+---@return teamTemplateData?
+function mockTeamTemplate.raw(teamtemplate, date)
+	assert(type(teamtemplate) == 'string', MISSING_INPUT_ERROR)
+	local cleanedInput = teamtemplate:lower():gsub('_', ' ')
+	local found = MockData[cleanedInput]
+	if not found then return end
+
+	if found.templatename then
+		return found
+	end
+
+	local historicals = {}
+	for startDate, teamTemplateName in pairs(found) do
+		table.insert(historicals, {startDate = startDate, name = teamTemplateName})
+	end
+	table.sort(historicals, function(a, b) return a.startDate < b.startDate end)
+
+	-- can not use DateExt.toYmdInUtc as mw.getContentLanguage():formatDate is not mocked
+	local standardizedDate
+	if type(date) == 'number' then
+		standardizedDate = os.date('%F', date)
+	else
+		standardizedDate = String.nilIfEmpty(date) or os.date('%F') --[[@as string]]
+	end
+
+	for index, info in ipairs(historicals) do
+		local endDate = (historicals[index + 1] or {}).startDate or os.date('%F')
+		if standardizedDate >= info.startDate and standardizedDate <= endDate then
+			return MockData[info.name]
```

Comment:
```suggestion
			-- add the historicaltemplate field
			foundCopy = Table.copy(MockData[info.name])
			foundCopy.historicaltemplate = cleanedInput
			return foundCopy 
```

Reply:
no, if it gets called with lets say `tl` it would set it as `tl` whereas it should be `team liquid`

Reply:
doing it in the data via
```lua
for alias, key in pairs(aliases) do
	data[alias] = Table.copy(data[key])
end
```
will change it and move the alias/redirect processing into the mock itself

Reply:
and needs some nil catching^^